### ISSUE TITLE
(HUNTER) Deterrence

### ISSUE THREAD

**Issue by UsulxD** [SunwellTracker/issues#1050]
Decription:  Detterence is not deflecting as it should.

How it works: Spells that are about to be casted or in the air for the first 0.3 ms are going through detterence

How it should work: Spells about to be casted or in the air for the first 0.3 ms should be completely deflected.

Source (you should point out proofs of your report, please give us some source):  Simply ask a warlock to mortal coil you, detterence will go up while you are horrified. Check AT for comparision.

**Comment by Akha72** [comment]
Proof?

**Comment by UsulxD** [comment]
@Akha72 I have no video software.  But there is a reason so little hunters are around in arena on both Feronis/Angrathar compared to all others 3.3.5s

**Comment by ujellyx** [comment]
Projectiles/spells mid-air shouldn't be deflected by Deterrence. It never worked that way.
Find us a video-proof, because just saying "check AT" doesn't proof anything.

**Comment by presstmoded** [comment]
![deterrence_bug](https://user-images.githubusercontent.com/21221088/39968352-b60683d6-56cb-11e8-96b7-288387b0fc05.jpg)

A proof that Deterrence sometimes bugs out (no idea what triggers it). I used it against warlock, after I've got hit by Immolate, and before I got hit by anything by the rogue, yet - after around 1 second - he managed to Pyro Rocket me (it could be "mid-air" as you like to explain, but rogue was in melee range and used that after my Deterrence), then Cheap Shot me and hit with some melee swings. Everything is included in the combat log screenshot I'm attaching to this post. Don't have any video though, but the buff aura of Deterrence was on, and the effect was visible all the time. I had this buff on till the very last moment I died.
Can't say what's the reason for that bug, it usually works just fine (Deadly Throw missed on me, and Conflagrate was deflected), and sometimes it suddenly stops working making pvp (and I believe pve in some cases) broken def-wise for this class. Was it an enough proof for you?
